cmake/modules/platform/toolcfg/openxl.cmake

```diff
@@ -0,0 +1,232 @@
+	set(OMR_C_WARNINGS_AS_ERROR_FLAG -qhalt=w)
+	set(OMR_CXX_WARNINGS_AS_ERROR_FLAG -qhalt=w)
+
+	set(OMR_C_ENHANCED_WARNINGS_FLAG )
+	set(OMR_CXX_ENHANCED_WARNINGS_FLAG )
+
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+		-qxflag=LTOL:LTOL0
+	)
+
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
+
+	if(OMR_ENV_DATA64)
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-m64
+		)
+	else()
+		# -march should be there for 32 and 64 C/CXX flags but the C compiler is used for
+		# the assembler and it has trouble with some assembly files if it is specified.
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-march=ppc
+		)
+	endif()
+
+	# Testarossa build variables. Longer term the distinction between TR and the rest
+	# of the OMR code should be heavily reduced. In the mean time, we keep
+	# the distinction.
+
+	# TR_COMPILE_OPTIONS are variables appended to CMAKE_{C,CXX}_FLAGS, and so
+	# apply to both C and C++ compilations.
+	list(APPEND TR_COMPILE_OPTIONS
+		-march=pwr7
+		-qtls
+		-qfuncsect
+		-qsuppress=1540-1087:1540-1088:1540-1090:1540-029:1500-029
+		-qdebug=nscrep
+	)
+
+	if(NOT CMAKE_C_COMPILER_IS_XLCLANG)
+		list(APPEND TR_COMPILE_OPTIONS
+			-qnotempinc
+			-qenum=small
+			-qmbcs
+		)
+	endif()
+
+	# Configure the platform dependent library for multithreading.
+	set(OMR_PLATFORM_THREAD_LIBRARY -lpthread)
+endif()
+
+if(OMR_OS_AIX)
+	list(APPEND OMR_PLATFORM_C_COMPILE_OPTIONS -qlanglvl=extended)
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
+
+	if(CMAKE_C_COMPILER_IS_XLCLANG)
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -qxlcompatmacros)
+	else()
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -qinfo=pro)
+	endif()
+
+	set(CMAKE_CXX_STANDARD_LIBRARIES "${CMAKE_CXX_STANDARD_LIBRARIES} -lm -liconv -ldl -lperfstat")
+
+	if(OMR_ENV_DATA64)
+		set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -q64")
+		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -q64")
+		set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -q64")
+
+		set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> -X64 cr <TARGET> <LINK_FLAGS> <OBJECTS>")
+		set(CMAKE_C_ARCHIVE_CREATE "<CMAKE_AR> -X64 cr <TARGET> <LINK_FLAGS> <OBJECTS>")
+		set(CMAKE_C_ARCHIVE_FINISH "<CMAKE_RANLIB> -X64 <TARGET>")
+	endif()
+
+elseif(OMR_OS_LINUX)
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+		-qxflag=selinux
+	)
+elseif(OMR_OS_ZOS)
+	set(OMR_ZOS_COMPILE_ARCHITECTURE "arch10" CACHE STRING "z/OS compile machine architecture" FORCE)
+	set(OMR_ZOS_COMPILE_TARGET "ZOSV2R4" CACHE STRING "z/OS compile target operating system" FORCE)
+	set(OMR_ZOS_COMPILE_TUNE "12" CACHE STRING "z/OS compile machine architecture tuning" FORCE)
+	set(OMR_ZOS_LINK_COMPAT "ZOSV2R4" CACHE STRING "z/OS link compatible operating system" FORCE)
+	set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "--shared")
+	set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "--shared")
+
+	set(CMAKE_ASM_FLAGS "-fno-integrated-as")
+	string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mgoff\"")
+	string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mSYSPARM(BIT64)\"")
```

Comment:
Shouldn't `BIT64` be conditional on `OMR_ENV_DATA64`?

Reply:
Added the missing conditional